### DISCUSSION TITLE
Can't use two apc ups units at once

### DISCUSSION THREAD

**Discussion by jatgm1** [hassio-addons/app-nut#202] [Q&A]
# Problem/Motivation
When adding two devices it only shows one device despite the integrations page showing two instances and two devices
> (Why the issue was filed)

## Expected behavior
Having two ups units
> (What you expected to happen)

## Actual behavior
Having one ups units that is mislabeled as another
> (What actually happened)

## Steps to reproduce
Add two ups's with the default configuration on auto in configuration in the addon in supervisor in home assistant
> (How can someone else make/see it happen)

## Proposed changes
I have no idea
> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

**Comment by sinclairpaul** [comment]
Post logs and config.

Don't use "auto" for multiple devices, you will need to set config as per the docs https://github.com/hassio-addons/addon-nut/blob/main/nut/DOCS.md#sub-option-config

**Comment by jatgm1** [comment]
How do I find a vendor Id or something identifying and then like specify it? Is there an option for finding something to use to identify it?

**Comment by jatgm1** [comment]
Both devices are literally basically identical so I think I need something to identify them

**Comment by sinclairpaul** [comment]
https://github.com/hassio-addons/addon-nut/blob/main/nut/DOCS.md#option-list_usb_devices

**Comment by jatgm1** [comment]
Bus 001 Device 004: ID 051d:0002 American Power Conversion Uninterruptible Power Supply
Bus 001 Device 003: ID 10c4:8a2a Silicon Labs HubZ Smart Home Controller
Bus 001 Device 005: ID 051d:0002 American Power Conversion Uninterruptible Power 

So they both have the same "Id" so how do I identify it? Is it possible to use the device number or whatever that is?

**Reply by sinclairpaul** [reply]
Specify the ports in the config "/dev/ttyS0" where 0 is the port, you will need to identify these yourself, and can likely be viewed from the supervisor/hardware tab.

**Comment by jatgm1** [comment]
1-1.2
Subsystem:
usb
Device path:
/dev/bus/usb/001/005
Attributes:
BUSNUM: '001'
DEVNAME: /dev/bus/usb/001/005
DEVNUM: '005'
DEVPATH: /devices/platform/soc/ffe09000.usb/ff500000.usb/xhci-hcd.4.auto/usb1/1-1/1-1.2
DEVTYPE: usb_device
DRIVER: usb
ID_BUS: usb
ID_FOR_SEAT: usb-platform-xhci-hcd_4_auto-usb-0_1_2
ID_MODEL: Back-UPS_ES_600M1_FW:928.a8_.D_USB_FW:a8
ID_MODEL_ENC: Back-UPS\x20ES\x20600M1\x20FW:928.a8\x20.D\x20USB\x20FW:a8\x20
ID_MODEL_ID: '0002'
ID_PATH: platform-xhci-hcd.4.auto-usb-0:1.2
ID_PATH_TAG: platform-xhci-hcd_4_auto-usb-0_1_2
ID_REVISION: '0090'
ID_SERIAL: >-
  American_Power_Conversion_Back-UPS_ES_600M1_FW:928.a8_.D_USB_FW:a8_4B2008P10626
ID_SERIAL_SHORT: 4B2008P10626
ID_USB_INTERFACES: ':030000:'
ID_VENDOR: American_Power_Conversion
ID_VENDOR_ENC: American\x20Power\x20Conversion
ID_VENDOR_ID: 051d
MAJOR: '189'
MINOR: '4'
PRODUCT: 51d/2/90
SUBSYSTEM: usb
TAGS: ':seat:'
TYPE: 0/0/0
USEC_INITIALIZED: '163240530667'
1-1.3
1-1.4
Subsystem:
usb
Device path:
/dev/bus/usb/001/004
Attributes:
BUSNUM: '001'
DEVNAME: /dev/bus/usb/001/004
DEVNUM: '004'
DEVPATH: /devices/platform/soc/ffe09000.usb/ff500000.usb/xhci-hcd.4.auto/usb1/1-1/1-1.4
DEVTYPE: usb_device
DRIVER: usb
ID_BUS: usb
ID_FOR_SEAT: usb-platform-xhci-hcd_4_auto-usb-0_1_4
ID_MODEL: Back-UPS_ES_600M1_FW:928.a8_.D_USB_FW:a8
ID_MODEL_ENC: Back-UPS\x20ES\x20600M1\x20FW:928.a8\x20.D\x20USB\x20FW:a8\x20
ID_MODEL_ID: '0002'
ID_PATH: platform-xhci-hcd.4.auto-usb-0:1.4
ID_PATH_TAG: platform-xhci-hcd_4_auto-usb-0_1_4
ID_REVISION: '0090'
ID_SERIAL: >-
  American_Power_Conversion_Back-UPS_ES_600M1_FW:928.a8_.D_USB_FW:a8_4B2008P10620
ID_SERIAL_SHORT: 4B2008P10620
ID_USB_INTERFACES: ':030000:'
ID_VENDOR: American_Power_Conversion
ID_VENDOR_ENC: American\x20Power\x20Conversion
ID_VENDOR_ID: 051d
MAJOR: '189'
MINOR: '3'
PRODUCT: 51d/2/90
SUBSYSTEM: usb
TAGS: ':seat:'
TYPE: 0/0/0
USEC_INITIALIZED: '3686514'

**Reply by jatgm1** [reply]
I found them but nothing was under tty anything

**Reply by jatgm1** [reply]
So I use the numbers 4 and 5 because it says 4 and 5 at the end of the device path and I'm not sure what else to use for port but the same thing happened again after everything was removed, adding a new device didn't work.

**Reply by jatgm1** [reply]
It added one, then the other, but after adding a second one the first one disappears

**Comment by jatgm1** [comment]
Dude idk what I'm looking for here but all the tty things don't have anything that says 051d or anything that says America power converter or apc or ups or anything. These look like the right ones though and idk what the port is for them.

**Comment by jatgm1** [comment]
So I'm kinda surprised theirs no way of like automatically searching for a ups unit or something because I cannot find out what to use for two identical ups units so that I may use them both at the same time in the same nut service running on a odroid n2+, I mean I'm really trying to monitor my servers power usage and the remaining battery so I can shut down high power devices after a certain amount of power being used, and I really kinda need this to work right and I'm not sure what I'm doing here. Is there a way to make like an auto detection thing? Because that would be super useful.

**Comment by sinclairpaul** [comment]
The NUT documentation discussing this driver is at https://networkupstools.org/docs/man/usbhid-ups.html

See the part Implementation

Unfortunately there isn't anything that can be done in the addon, it wraps the NUT software

**Reply by jatgm1** [reply]
Okay but like how do I find any of the information i can use to distinguish between the two ups units? Is there like no way of doing this?

**Reply by jatgm1** [reply]
Like to use two ups units on the same nut server that are the same units? I mean I kinda wanted some consistency so I got the same ones

**Reply by sinclairpaul** [reply]
The documentation states the following for identifiers "vendor", "product", "serial", "vendorid", and "productid".

I'd suggest looking at the serial option.  In all honesty this really isn't an issue with the addon, but with NUT configuration itself.

The only suggestion I can make is you set:

serial - *4B2008P10620

and 

serial - *4B2008P10626

But you will need to experiment with your own setup, or alternatively post me your ups devices so I can test 😉

**Reply by jatgm1** [reply]
Oh crap I was using vendor Id, I thought that was what it said to do. I think it's working
